http.go

```diff
@@ -0,0 +1,213 @@
+			return errors.Wrap(err, "decode payload to json")
+		}
+		return nil
+	}
+}
+
+// httpClient is an interface which abstracts behaviors that the Cloud Go SDK
+// will perform.
+type httpClient interface {
+	sendGetRequest(path, query string, payloadDecodeFunc payloadDecodeFunc) error
+	sendPostRequest(path, query string, body interface{}, payloadDecodeFunc payloadDecodeFunc) error
+	sendPutRequest(path, query string, body interface{}, payloadDecodeFunc payloadDecodeFunc) error
+	sendDeleteRequest(path, query string, payloadDecodeFunc payloadDecodeFunc) error
+	sendRequest(req *http.Request, payloadDecodeFunc payloadDecodeFunc) error
+}
+
+type httpClientImpl struct {
+	url    *url.URL
+	token  *AccessToken
+	client *http.Client
+}
+
+func constructHTTPClient(opts *Options, token *AccessToken) (httpClient, error) {
+	u, err := url.Parse(opts.ServerAddr)
+	if err != nil {
+		return nil, errors.Wrap(err, "construct http client")
+	}
+
+	tr := &http.Transport{
+		DialContext: func(ctx context.Context, network string, addr string) (net.Conn, error) {
+			return net.DialTimeout(network, addr, opts.DialTimeout)
+		},
+
+		TLSClientConfig: &tls.Config{
+			ServerName:         opts.ServerNameIndication,
+			InsecureSkipVerify: opts.InsecureSkipTLSVerify,
+			MinVersion:         tls.VersionTLS11,
+			MaxVersion:         tls.VersionTLS13,
+		},
+		TLSHandshakeTimeout: opts.TLSHandshakeTimeout,
+	}
+
+	return &httpClientImpl{
+		url: u,
+		client: &http.Client{
+			Transport: tr,
+		},
+		token: token,
+	}, nil
+}
+
+func (impl *httpClientImpl) sendGetRequest(path, query string, payloadDecodeFunc payloadDecodeFunc) error {
+	u := *impl.url
+	u.Path = path
+	u.RawQuery = query
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return errors.Wrap(err, "construct http request")
+	}
+	return impl.sendRequest(req, payloadDecodeFunc)
+}
+
+func (impl *httpClientImpl) sendPostRequest(path, query string, body interface{}, payloadDecodeFunc payloadDecodeFunc) error {
+	u := *impl.url
+	u.Path = path
+	u.RawQuery = query
+
+	data, err := json.Marshal(body)
+	if err != nil {
+		return errors.Wrap(err, "encode http request body")
+	}
+
+	req, err := http.NewRequest(http.MethodPost, u.String(), bytes.NewReader(data))
+	if err != nil {
+		return errors.Wrap(err, "construct http request")
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	return impl.sendRequest(req, payloadDecodeFunc)
+}
+
+func (impl *httpClientImpl) sendPutRequest(path, query string, body interface{}, payloadDecodeFunc payloadDecodeFunc) error {
+	u := *impl.url
+	u.Path = path
+	u.RawQuery = query
+
+	data, err := json.Marshal(body)
+	if err != nil {
+		return errors.Wrap(err, "encode http request body")
+	}
+
+	req, err := http.NewRequest(http.MethodPut, u.String(), bytes.NewReader(data))
+	if err != nil {
+		return errors.Wrap(err, "construct http request")
+	}
+
+	req.Header.Set("Content-Type", "application/json")
```

Comment:
Is it possible to handle it in sendRequest uniformly?

Reply:
`sendRequest` doesn't know the type of request body.